sound/soc/sof/imx/imx8.c

```diff
@@ -116,7 +117,19 @@ static void imx8_dsp_handle_request(struct imx_dsp_ipc *ipc)
 {
 	struct imx8_priv *priv = imx_dsp_get_data(ipc);
 
-	snd_sof_ipc_msgs_rx(priv->sdev);
+	int p; /* panic code */
+
+	/* Read the message from the debug box. */
+	sof_mailbox_read(priv->sdev, priv->sdev->debug_box.offset + 4, &p, 4);
+
+	/* Check to see if the message is a panic code (0x0dead***) */
+	if ((p & SOF_IPC_PANIC_MAGIC_MASK) == SOF_IPC_PANIC_MAGIC) {
+		/* Start panic routine */
+		snd_sof_dsp_panic(priv->sdev, p);
+	} else {
+		/* If it is not a panic code, reply to the message. */
+		snd_sof_ipc_msgs_rx(priv->sdev);
+	}
```

Comment:
you actually don't need those braces there

Reply:
It's two lines (code + comment), I would actually prefer the braces for this one. Unnecessary but I honestly would recommend them.